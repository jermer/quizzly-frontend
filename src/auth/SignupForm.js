import { useState } from 'react';
import { useHistory } from 'react-router-dom';

const SignupForm = ({ signup }) => {
    const history = useHistory();
    const [formData, setFormData] = useState({
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: ""
    });
    const [formErrors, setFormErrors] = useState([]);

    async function handleSubmit(evt) {
        evt.preventDefault();
        const result = await login(formData);
        if (result.success) {
            // redirect to the homepage
            history.push("/");
        } else {
            setFormErrors(result.errors);
        }
    };

    function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData(fdata => ({ ...fdata, [name]: value }));
    };

    /** Render form */
    return (
        <form onSubmit={handleSubmit}>
            <p>Under construction...</p>
        </form>
    );
}

export default SignupForm;
