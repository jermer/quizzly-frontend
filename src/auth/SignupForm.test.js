import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import SignupForm from './SignupForm';

// smoke test
it('renders without crashing', () => {
    render(
        <MemoryRouter>
            <SignupForm />
        </MemoryRouter>
    );
});

// snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <SignupForm />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
