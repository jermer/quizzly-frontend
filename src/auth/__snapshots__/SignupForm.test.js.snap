// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h3
        class="mb-3"
      >
        Sign Up
      </h3>
      <div
        class="card"
      >
        <div
          class="card-body text-start"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="username"
                class="form-control"
                id="username"
                name="username"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                autocomplete="email"
                class="form-control"
                id="email"
                name="email"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="form-control"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="text-center mt-3"
            >
              <button
                class="btn btn-primary float-right"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
